src/blog/2025-08-18-oxlint-type-aware.md

```diff
@@ -0,0 +1,258 @@
+
+We determined that writing our own type-inferencer or type-checker was not feasible due to
+the challenge of keeping up with a fast-moving target like TypeScript.
+
+### Communication with TypeScript Compiler
+
+Prior to `typescript-go`, projects added plugin interfaces to TypeScript's public API by either mapping its AST to `estree` or directly traversing the TypeScript AST. Examples include:
+
+- [typescript-eslint](https://typescript-eslint.io/getting-started/typed-linting)
+- [tsslint](https://github.com/johnsoncodehk/tsslint)
+- [tsl](https://github.com/ArnaudBarre/tsl)
+
+We also explored [inter-process communication with oxlint](https://github.com/oxc-project/oxc/discussions/2855) but abandoned the idea.
+
+With `typescript-go`, the TypeScript team is [leaning towards](https://github.com/microsoft/typescript-go/discussions/455)
+encoding the TypeScript AST and decoding it on the JavaScript side through inter-process communication.
+
+While these approaches work, they still incur:
+
+- Performance issues of varying degrees that don't suit oxlint's performance characteristics.
+- The cost of maintaining an AST mapping from TypeScript's AST.
+
+## Considerations
+
+While `tsgolint` solves the performance issue, there are other technical challenges that need to be addressed.
+
+### Requirement for a Different TypeScript Version
+
+We plan to release snapshots of `typescript-go` versions and align their version numbers with TypeScript.
+You will then be able to install `oxlint-typescript` with the correct TypeScript version.
+
+The downside of this approach is that you may need to upgrade TypeScript if `oxlint-tsgolint` requires changes.
+
+### Maintenance cost of `tsgolint`
+
+Shimming TypeScript's internal APIs carries some risk. However, the TypeScript AST and its visitor are actually quite stable.
+We accept this risk and will fix breaking changes when upgrading `typescript-go`.
+
+Our `typescript-go` version is synced many times a week.
+
+## Performance Issues
+
+`tsgolint` currently does not perform well on large monorepos with hundreds of projects or lots of project references.
+
+It may hang or drain memory if a bug is encountered.
+
+We are actively profiling and submitting improves to `typescript-go`, benefiting all `typescript-go` users.
+
+Our core team member [@camc314](https://github.com/camc314) has already submitted [many PRs](https://github.com/microsoft/typescript-go/pulls?q=is%3Apr+author%3Acamc314+) that made several code paths significantly faster.
+
+## Acknowledgements
+
+We'd like to extend our gratitude to:
+
+- The TypeScript team for creating `typescript-go`.
+- The `typescript-eslint` team for their heartwarming support.
+- [@auvred](https://github.com/auvred) for creating `tsgolint`.
+- [@camchenry](https://github.com/camchenry) for the `oxlint` + `tsgolint` integration.
+- [@camc314](https://github.com/camc314) for triaging performance issues.
```

Comment:
```suggestion
- [@camc314](https://github.com/camc314) for work on performance issues.
```

"triaging" underplays his contribution. He hasn't just investigated the issues, he's solved many of them!